symengine/parser/sbml/sbml_parser.h

```diff
@@ -0,0 +1,41 @@
+#ifndef SYMENGINE_PARSER_SBML_PARSER_H
+#define SYMENGINE_PARSER_SBML_PARSER_H
+
+#include <fstream>
+#include <algorithm>
+#include <memory>
+
+#include <symengine/parser/sbml/sbml_tokenizer.h>
+#include <symengine/add.h>
+#include <symengine/pow.h>
+#include <symengine/logic.h>
+
+namespace SymEngine
+{
+
+class SbmlParser
```

Comment:
Can you make this inherit from the string parse so that `parse_numeric` and other functions need not be duplicated?

Reply:
Yes - after struggling with strange compiler errors it turns out this also required a change to the main parser: https://github.com/symengine/symengine/pull/1785/commits/b9d40f898e83b5bb6e030198fae5a863974c16d0